digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -172,7 +172,16 @@ class ProjectService(roadAddressService: RoadAddressService, roadLinkService: Ro
     if(linksInProject.nonEmpty){
       ProjectDAO.removeProjectLinksByProjectAndRoadNumber(roadAddressProjectID, newRoadNumber, newRoadPartNumber)
     }
-    val randomSideCode = SideCode.TowardsDigitizing
+    val randomSideCode =
+      linksInProject.map(l => l -> projectAddressLinks.find(n => GeometryUtils.areAdjacent(l.geometry, n.geometry))).toMap.find { case (l, n) => n.nonEmpty }.map {
+        case (l, Some(n)) =>
+          if (GeometryUtils.areAdjacent(l.geometry.head, n.geometry.last) ||
+            GeometryUtils.areAdjacent(l.geometry.last, n.geometry.head))
+            l.sideCode
+          else
+            switchSideCode(l.sideCode)
+        case _ => SideCode.TowardsDigitizing
+      }.getOrElse(SideCode.TowardsDigitizing)
```

Comment:
could this "OrElse(SideCode.TowardsDigitizing)" be redundant since we already have "case _ => SideCode.TowardsDigitizing"?